sycl/plugins/unified_runtime/pi2ur.hpp

```diff
@@ -801,4 +841,134 @@ inline pi_result piDevicePartition(
+      {PI_EXT_CONTEXT_INFO_ATOMIC_MEMORY_SCOPE_CAPABILITIES,
+       (ur_context_info_t)UR_EXT_CONTEXT_INFO_ATOMIC_MEMORY_SCOPE_CAPABILITIES},
+      {PI_EXT_ONEAPI_CONTEXT_INFO_USM_MEMCPY2D_SUPPORT,
+       UR_CONTEXT_INFO_USM_MEMCPY2D_SUPPORT},
+      {PI_EXT_ONEAPI_CONTEXT_INFO_USM_FILL2D_SUPPORT,
+       UR_CONTEXT_INFO_USM_FILL2D_SUPPORT},
+      {PI_EXT_ONEAPI_CONTEXT_INFO_USM_MEMSET2D_SUPPORT,
+       (ur_context_info_t)UR_EXT_CONTEXT_INFO_USM_MEMSET2D_SUPPORT},
+  };
+
+  auto InfoType = InfoMapping.find(param_name);
+  if (InfoType == InfoMapping.end()) {
+    return PI_ERROR_UNKNOWN;
+  }
+
+  size_t SizeInOut = param_value_size;
+  auto hContext = reinterpret_cast<ur_context_handle_t>(context);
+  HANDLE_ERRORS(urContextGetInfo(hContext, InfoType->second, SizeInOut,
+                                 param_value, param_value_size_ret));
+  fixupInfoValueTypes(SizeInOut, param_value_size_ret, param_value);
+
+  return PI_SUCCESS;
+}
+
+inline pi_result piContextRetain(pi_context context) {
+  auto hContext = reinterpret_cast<ur_context_handle_t>(context);
+  HANDLE_ERRORS(urContextRetain(hContext));
+
+  return PI_SUCCESS;
+}
+
+inline pi_result piContextRelease(pi_context context) {
+  auto hContext = reinterpret_cast<ur_context_handle_t>(context);
+  HANDLE_ERRORS(urContextRelease(hContext));
+
+  return PI_SUCCESS;
+}
+
+inline pi_result piextDeviceGetNativeHandle(pi_device device,
+                                            pi_native_handle *nativeHandle) {
+  auto hDevice = reinterpret_cast<ur_device_handle_t>(device);
+  auto phNativeHandle = reinterpret_cast<ur_native_handle_t *>(nativeHandle);
+
+  HANDLE_ERRORS(urDeviceGetNativeHandle(hDevice, phNativeHandle));
+
+  return PI_SUCCESS;
+}
+
+inline pi_result
+piextDeviceCreateWithNativeHandle(pi_native_handle nativeHandle,
+                                  pi_platform platform, pi_device *device) {
+  auto hNativeHandle = reinterpret_cast<ur_native_handle_t>(nativeHandle);
+  auto hPlatform = reinterpret_cast<ur_platform_handle_t>(platform);
+  auto phDevice = reinterpret_cast<ur_device_handle_t *>(device);
+
+  HANDLE_ERRORS(
+      urDeviceCreateWithNativeHandle(hNativeHandle, hPlatform, phDevice));
+
+  return PI_SUCCESS;
+}
+
+inline pi_result piextContextSetExtendedDeleter(
+    pi_context context, pi_context_extended_deleter func, void *user_data) {
+  auto hContext = reinterpret_cast<ur_context_handle_t>(context);
+
+  HANDLE_ERRORS(urContextSetExtendedDeleter(hContext, func, user_data));
+
+  return PI_SUCCESS;
+}
+
+inline pi_result piextContextGetNativeHandle(pi_context context,
+                                             pi_native_handle *nativeHandle) {
+  auto hContext = reinterpret_cast<ur_context_handle_t>(context);
+  auto phNativeHandle = reinterpret_cast<ur_native_handle_t *>(nativeHandle);
+
+  HANDLE_ERRORS(urContextGetNativeHandle(hContext, phNativeHandle));
+
+  return PI_SUCCESS;
+}
+
+inline pi_result piextContextCreateWithNativeHandle(

```

Comment:
Please see implementation used for UR L0 here, https://github.com/jandres742/llvm/blob/b9a7fbb4da41c8c09645011a9f6ff926fba77725/sycl/plugins/unified_runtime/pi2ur.hpp#L1034. That would also work here, right?